.github/actions/deploy_docs/action.yml

```diff
@@ -0,0 +1,46 @@
+name: 'Deploy documentation'
+description: 'Deploy ESPResSo documentation'
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        # checkout github pages
+        cd "${HOME}"
+        git clone --quiet git@github.com:espressomd/espressomd.github.io.git
+        cd espressomd.github.io
+        LAST_COMMIT=$(git log -1 --pretty=format:"%s" remotes/origin/main -- doc dox tutorials)
+        NEXT_COMMIT="Documentation for ${GITHUB_SHA}"
+        if [ "${NEXT_COMMIT}" = "${LAST_COMMIT}" ]; then
+          echo "Documentation already up-to-date.";
+          exit 0;
+        fi
+        # download artifacts
+        curl https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/artifacts/python/download?job=run_tutorials -L -o tutorials.zip
+        curl https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/artifacts/python/download?job=run_doxygen   -L -o doxygen.zip
+        curl https://gitlab.icp.uni-stuttgart.de/espressomd/espresso/-/jobs/artifacts/python/download?job=check_sphinx  -L -o sphinx.zip
+        if grep -F '<!DOCTYPE html>' *.zip; then
+          echo "The artifacts could not be downloaded.";
+          exit 1;
```

Comment:
Are you sure that's enough to verify that all 3 downloads worked?

Reply:
We can add the `curl` flag `--fail`, which prints out the HTTP error code in case of failure. The `curl` man pages say it's not working on all HTTP error codes, but I tried a few common ones (typo in url and in GET query) and it printed out the expected 404 error code instead of downloading the fancy 404 HTML page, which is helpful for debugging.